subsys/shell/shell_history.c

```diff
@@ -87,6 +98,17 @@ void shell_history_put(struct shell_history *history, u8_t *line, size_t len)
 {
 	sys_dnode_t *l_item; /* list item */
 	struct shell_history_item *h_item;
+	u32_t total_len = len + offsetof(struct shell_history_item, data);
+	u32_t claim_len;
+	u32_t claim2_len;
+	u16_t padding = (~total_len + 1) & 3;
```

Comment:
need this to be (sizeof(void *) - 1) instead of 3.  This covers 32 bit and 64 bit, which is coming in the future (intel)

Reply:
done